core/test/prediction/SurvivalPredictionStrategyTest.cpp

```diff
@@ -0,0 +1,156 @@
+/*-------------------------------------------------------------------------------
+  This file is part of generalized random forest (grf).
+
+  grf is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  grf is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with grf. If not, see <http://www.gnu.org/licenses/>.
+ #-------------------------------------------------------------------------------*/
+
+#include "commons/Data.h"
+#include "commons/utility.h"
+#include "prediction/SurvivalPredictionStrategy.h"
+
+#include "catch.hpp"
+
+using namespace grf;
+
+TEST_CASE("Kaplan-Meier survival estimates are correct", "[survival], [prediction]") {
+  /*
```

Comment:
We could move this into a README too, like we do for splitting and characterization tests?